### ISSUE TITLE
error: failed to legalize operation 'ttir.pooling' that was explicitly marked illegal

### ISSUE THREAD

**Issue by mmilosevicTT** [tenstorrent/tt-mlir#6006]
### Describe the bug

Several versions of `inception` TT-XLA PyTorch model are failing in training with the `ttir.pooling`.

Sample output:
- [output_mlir.txt](https://github.com/user-attachments/files/23723025/output1_mlir.txt)

From all tests the same ttir op is problematic:
```
%891 = "ttir.pooling"(%34, %890) <{base_dilations = array<i64: 1, 1>, operandSegmentSizes = array<i32: 1, 1>, padding = array<i64: 0, 0, 0, 0>, pooling_method = #ttir<pooling_method Sum>, window_dilations = array<i64: 1, 1>, window_dimensions = array<i64: 3, 3>, window_strides = array<i64: 1, 1>}> : (tensor<10x10xbf16>, tensor<8x8xbf16>) -> tensor<8x8xbf16> loc(#loc1031)
```

### Steps to reproduce the issue

1. Go to TT-XLA
2. Replace NOT_SUPPORTED_SKIP tag with EXPECTED_PASSING in the `tests/runner/test_config/torch/test_config_training_single_device.yaml`
3. Run `pytest tests/runner/test_models.py::test_all_models_torch[<TEST_NAME>]`, or any other of the mentioned models.
4. Full list of models failing with this error is present in `tests/runner/test_config/torch/test_config_training_single_device.yaml`

**Comment by gengelageTT** [comment]
I've been running into the same error when running some models through tt-xla in data-parallel. In my case, the issue was that the TTIRFusing pass was failing to fuse the following pattern since the input had been reshaped and broadcast:
```
%9 = "ttir.full"() <{fill_value = 1.000000e+00 : f32, shape = array<i32>}> : () -> tensor<bf16>
%13 = "ttir.reshape"(%9, %12) <{shape = [1 : i32, 1 : i32]}> : (tensor<bf16>, tensor<1x1xbf16>) -> tensor<1x1xbf16>
%15 = "ttir.broadcast"(%13, %14) <{broadcast_dimensions = array<i64: 7, 7>}> : (tensor<1x1xbf16>, tensor<7x7xbf16>) -> tensor<7x7xbf16>
...
%6651 = "ttir.pooling"(%6649, %6650) <{base_dilations = array<i64: 1, 1, 1, 1>, operandSegmentSizes = array<i32: 1, 1>, padding = array<i64: 0, 0, 0, 0, 0, 0, 0, 0>, pooling_method = #ttir<pooling_method Sum>, window_dilations = array<i64: 1, 1, 1, 1>, window_dimensions = array<i64: 1, 1, 7, 7>, window_strides = array<i64: 1, 1, 1, 1>}> : (tensor<1x2048x7x7xbf16>, tensor<1x2048x1x1xbf16>) -> tensor<1x2048x1x1xbf16>
%6653 = "ttir.pooling"(%15, %6652) <{base_dilations = array<i64: 1, 1>, operandSegmentSizes = array<i32: 1, 1>, padding = array<i64: 0, 0, 0, 0>, pooling_method = #ttir<pooling_method Sum>, window_dilations = array<i64: 1, 1>, window_dimensions = array<i64: 7, 7>, window_strides = array<i64: 1, 1>}> : (tensor<7x7xbf16>, tensor<1x1xbf16>) -> tensor<1x1xbf16>
%6655 = "ttir.reshape"(%6653, %6654) <{shape = [1 : i32, 1 : i32, 1 : i32, 1 : i32]}> : (tensor<1x1xbf16>, tensor<1x1x1x1xbf16>) -> tensor<1x1x1x1xbf16>
%6657 = "ttir.broadcast"(%6655, %6656) <{broadcast_dimensions = array<i64: 1, 2048, 1, 1>}> : (tensor<1x1x1x1xbf16>, tensor<1x2048x1x1xbf16>) -> tensor<1x2048x1x1xbf16>
%6659 = "ttir.div"(%6651, %6657, %6658) : (tensor<1x2048x1x1xbf16>, tensor<1x2048x1x1xbf16>, tensor<1x2048x1x1xbf16>) -> tensor<1x2048x1x1xbf16>
```
I put up a pr that fixes my issue here: https://github.com/tenstorrent/tt-mlir/pull/6077. Hopefully it'll fix the issue you're seeing too.

**Comment by djstefanovicTT** [comment]
Should have been solved by https://github.com/tenstorrent/tt-mlir/pull/5975, the given example has two dimensional input. Ran a test with `%891 = "ttir.pooling"(%34, %890) <{base_dilations = array<i64: 1, 1>, operandSegmentSizes = array<i32: 1, 1>, padding = array<i64: 0, 0, 0, 0>, pooling_method = #ttir<pooling_method Sum>, window_dilations = array<i64: 1, 1>, window_dimensions = array<i64: 3, 3>, window_strides = array<i64: 1, 1>}> : (tensor<10x10xbf16>, tensor<8x8xbf16>) -> tensor<8x8xbf16> loc(#loc1031)` (example from the issue) converted to stableHLO and it got successfully validated and translated.
